com.unity.shadergraph/Editor/Data/Nodes/Input/Scene/LODFadeNode.cs

```diff
@@ -0,0 +1,41 @@
+using System.Linq;
+using UnityEngine;
+using UnityEditor.Graphing;
+using UnityEditor.ShaderGraph.Drawing.Controls;
+using UnityEditor.ShaderGraph.Internal;
+
+namespace UnityEditor.ShaderGraph
+{
+    [Title("Input", "Geometry", "LOD Fade")]
+    class LODFadeNode : AbstractMaterialNode, IGeneratesBodyCode
+    {
+        public const int OutputSlotFadeId = 0;
+        public const int OutputSlotQuantizedId = 1;
+        const string kOutputSlotFadeName = "Fade";
+        const string kOutputSlotQuantizedName = "Quantized Fade";
+
+
+        public override bool hasPreview { get { return false; } }
+
+        public LODFadeNode()
+        {
+            name = "LOD Fade";
+            synonyms = new string[] { "fade", "disolve", "cross fade", "blend", "level of detail" };
+            UpdateNodeAfterDeserialization();
+        }
+
+        public sealed override void UpdateNodeAfterDeserialization()
+        {
+            AddSlot(new Vector1MaterialSlot(OutputSlotFadeId, kOutputSlotFadeName, kOutputSlotFadeName, SlotType.Output, 0, ShaderStageCapability.All));
+            AddSlot(new Vector1MaterialSlot(OutputSlotQuantizedId, kOutputSlotQuantizedName, kOutputSlotQuantizedName, SlotType.Output, 0, ShaderStageCapability.All));
+            RemoveSlotsNameNotMatching(new[] { OutputSlotFadeId, OutputSlotQuantizedId });
+        }
+
+        // Node generations
+        public virtual void GenerateNodeCode(ShaderStringBuilder sb, GenerationMode generationMode)
+        {
+            sb.AppendLine(string.Format("$precision {0} = unity_LODFade.x;", GetVariableNameForSlot(OutputSlotFadeId)));
+            sb.AppendLine(string.Format("$precision {0} = unity_LODFade.y;", GetVariableNameForSlot(OutputSlotQuantizedId)));
```

Comment:
Do we know where unity_LODFade is defined? Is it always set/defined for all unity shaders?

Reply:
Unity_LODFade is define in the UnityPerDraw.
This is currently a good question.  I guess it is always define in the context of shader graph (even for procedural geometry) as the other values like unity_SHAr. Doesn't mean the content of the value will make sense.

Reply:
We don't have a way to guarantee that this node will handle situations where it isn't present-- but that's a separate problem. LGTM.